### ISSUE TITLE
BUG: TypeError: protocols.has is not a function (5.6.1 version)

### ISSUE THREAD

**Issue by hiepxanh** [enisdenjo/graphql-ws#318]
I got this bug on new version

![image](https://user-images.githubusercontent.com/16970990/155335633-50c07248-9ef4-4b4f-8881-b1d21e77fa22.png)

```
handleProtocol?? [ 'graphql-transport-ws' ] graphql-transport-ws
/batcave/node_modules/graphql-ws/lib/server.js:280
        : protocols.has(common_1.GRAPHQL_TRANSPORT_WS_PROTOCOL))
                    ^
TypeError: protocols.has is not a function
    at Object.handleProtocols (/batcave/node_modules/graphql-ws/lib/server.js:280:21)
    at WebSocketServer.completeUpgrade (/batcave/node_modules/ws/lib/websocket-server.js:323:33)
    at WebSocketServer.handleUpgrade (/batcave/node_modules/ws/lib/websocket-server.js:271:10)
    at Server.<anonymous> (/batcave/node_modules/postgraphile/build/postgraphile/http/subscriptions.js:369:21)
    at Server.emit (node:events:520:28)
    at onParserExecuteCommon (node:_http_server:727:14)
    at onParserExecute (node:_http_server:641:3)

```

**Comment by enisdenjo** [comment]
[`has` does exist on a `Set`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/has), which is what I am [expecting](https://github.com/enisdenjo/graphql-ws/blob/96058eb08f9c98ea4102f0aa90cc25f7be12f955/src/server.ts#L865) there.

Which version of [`ws`](https://github.com/websockets/ws) are you using?

**Comment by enisdenjo** [comment]
Seems like the [`protocols` argument in `WebSocketServer.handleProtocols` was an Array in v7](https://github.com/websockets/ws/blob/v7.x/lib/websocket-server.js#L317-L323) ([also stated in the breaking changes of v8](https://github.com/websockets/ws/releases/tag/8.0.0)), the fix in [v5.6.2](https://github.com/enisdenjo/graphql-ws/releases/tag/v5.6.2) covers the issue.

Additionally, I've made sure that all future [tests run on `ws@7`](https://github.com/enisdenjo/graphql-ws/commit/31247544463d64f5b0866cbe9cd86617179ca9ca) too so bugs like these won't creep up.

Thanks for reporting!

**Comment by hiepxanh** [comment]
ya, look like that is the root cause of this issue. Thank you for your quick fix, very nice <3
